packages/truffle/test/scenarios/commandRunner.js

```diff
@@ -71,6 +72,44 @@ module.exports = {
         config.logger.log("OUT: ", data);
       });
 
+      child.on("close", code => {
+        config.logger.log("EXIT: ", code);
+        resolve();
+      });
+    });
+  },
+  runInConsoleEnvironment: function (options, config) {
```

Comment:
`runInConsoleEnvironment` starts a child process and immediately closes it, which, doesn't really run anything. `runInDevelopEnvironment`, on the other hand, starts a child process based on a config, and issues a list of commands in the child process environment before exiting.

I think this current test could be satisfied if `runInDevelopEnvironment` were refactored. This would also open the door for more robust integration `truffle console` tests. Consider this (it is just a suggestion, and has pros/cons):

1. rename `runInDevelopEnvironment` to `runInREPL`
   ```javascript
    runInREPL(commands: string[], options: { replType: 'console' | 'develop', config: TruffleConfig, displayHost: string }) { /*...*/ }
    
    // develop test
    runInREPL(['migrate', 'networks'], { config, displayHost: 'develop', replType: 'develop' }):

    // console test
    const url = new URL(/* url-like-string */);
    runInREPL([], { config, displayHost:  url.host, replType: 'console' });   
2. update existing develop tests

Reply:
Thanks @cds-amal for the feedback. This make sense to have one function for both `develop` and `console` integration test. We can apply code reusability here. I will work on this. :)